### DISCUSSION TITLE
Suggestion: Allow auto-generated Enum.IsDefined parameter checks, similar to the current "Add null check" for reference types

### DISCUSSION THREAD

**Discussion by chance2now** [dotnet/roslyn#66326] [Ideas]
Currently, we can click on a parameter name, and if a null-check is appropriate for that parameter, we get an option "Add null check" that generates the code for the check. I wondered if a similar option could be added for enum-type parameters that would automatically generate code for an "enum defined" check. Basically, if we have a parameter typed as "`MyEnum param`", the suggestion would auto generate something like:

`        if (!Enum.IsDefined(param)) throw new InvalidEnumArgumentException(nameof(param), (int)param, param.GetType());`

Does this make sense to add as a suggestion?

**Comment by ryzngard** [comment]
I don't think it's a common use case, but you could add as a suggestion. I think we'd take a community PR for implementing this (with some feedback on how often it shows up) but unlikely we'd prioritize working on this without more traction.
